### ISSUE TITLE
reduntantParens does not strip all parens

### ISSUE THREAD

**Issue by proxi** [nicklockwood/SwiftFormat#357]
`reduntantParens` rules does not remove all parent it could, for example given input:

```
func foo() {
    if (1 == 1) && (2 == 2) {
        print("Ok!")
    }
}
```

output is:

```
func foo() {
    if 1 == 1, (2 == 2) {
        print("Ok!")
    }
}
```

where it should be:

```
func foo() {
    if 1 == 1, 2 == 2 {
        print("Ok!")
    }
}
```

**Comment by nicklockwood** [comment]
@proxi thanks for reporting. I'll fix it asap.

**Comment by nicklockwood** [comment]
@proxi fixed in 0.39.1.

**Comment by proxi** [comment]
Thank you @nicklockwood
